main.go

```diff
@@ -791,7 +797,9 @@ func stripTenantPrefix(prefix string, next http.Handler) http.Handler {
 			http.Error(w, "tenant not found", http.StatusInternalServerError)
 			return
 		}
-		http.StripPrefix(path.Join("/", prefix, tenant), next).ServeHTTP(w, r)
+
+		tenantPrefix := path.Join("/", prefix, tenant)
+		http.StripPrefix(tenantPrefix, next).ServeHTTP(w, r.Clone(context.WithValue(r.Context(), proxy.TenantPrefixKey, tenantPrefix)))
```

Comment:
Maybe we can reduce the complexity of this one (intense) line by using a `WithPrefix` middleware to set the tenant prefix as the prefix in the context

Reply:
Ah, there is one gotcha here though, since the prefix we are stripping is dynamic, we have to extract the `tenant` from the request again, and set the header (which is exactly what we are doing here, hence I kept it the same function).

Reply:
We do this all over the place though by writing the middleware instead as a middleware generator:

```go
func StripPrefixMiddleware(prefix string) func(next http.Handler) http.Handler {
    return func(next http.Handler) http.Handler {
        return func(w http.ResponseWriter, r *http.Request) {
            //something with prefix
        }
    }
}
```